src/main/java/com/capstone/doctorsOffice/entities/Doctor.java

```diff
@@ -77,5 +86,17 @@ public Set<Appointment> getAppointments() {
     public void setAppointments(Set<Appointment> appointments) {
         this.appointments = appointments;
     }
+
+    public Doctor(DoctorDto doctorDto) {
+        if (doctorDto.getId() != null) {
+            this.id = id;
+        }
```

Comment:
Its the dto value that should be assigned here.
should be:
```
if (doctorDto.getId() != null) {
             this.id = doctorDto.getId();
         }
```

Change the other two below as well.